crates/iceberg/src/io/file_io.rs

```diff
@@ -409,6 +418,8 @@ pub struct OutputFile {
     path: String,
     // Relative path of file to uri, starts at [`relative_path_pos`]
     relative_path_pos: usize,
+    // Whether to use append mode for writes (required for some storage backends like AZDLS)
+    append_file: bool,
```

Comment:
I'm generally not a fan of adding bools as args since it limits extension/API changes in the future and sets a precedent for exponentially complex input arguments that might not all be valid configs. Can we envision any other things that we might want to encode in a `WriteMode` enum with one value representing the current behavior and maybe an `Append` value. If the answer is no then I'll accept the `bool`, otherwise we might consider an enum.

Reply:
Thanks a lot for your comment. Using bool is indeed not a good practice. Let me refactor it to Enum. However, I'm not sure if we should expose this parameter externally for now, because in the bug I encountered, only AZDLS uses it, and this is related to opendal's implementation.